src/core/layout/qgslayoutitemchart.cpp

```diff
@@ -78,7 +78,12 @@ void QgsLayoutItemChart::setPlot( QgsPlot *plot )
     {
       Qgs2DXyPlot *oldPlot2dXy = dynamic_cast<Qgs2DXyPlot *>( mPlot.get() );
       Qgs2DXyPlot *newPlot2dXy = dynamic_cast<Qgs2DXyPlot *>( plot2d );
-      if ( oldPlot2dXy && oldPlot2dXy->xAxis().type() != newPlot2dXy->xAxis().type() )
+      if ( oldPlot2dXy && oldPlot2dXy->xAxis().type() == newPlot2dXy->xAxis().type() )
```

Comment:
There's a crash risk here -- newPlot2dXy may be nullptr if the new plot isn't a XY plot

Reply:
@nyalldawson , I don't think it's possible to have a crash here because of the condition on line 73 which insures that in this block we have the same type of plot for old and new.

Reply:
Hmm... true... but it's a tiny change and will make static analysis tools happier (or we'll get eg coverity "unchecked dynamic cast" warnings)

Reply:
OK done.